dsl/src/test/scala/org/http4s/dsl/PathInHttpRoutesSuite.scala

```diff
@@ -89,6 +90,11 @@ class PathInHttpRoutesSuite extends Http4sSuite {
     case GET -> Root / "flagparam" :? Flag(flag) =>
       if (flag) Ok("flag present")
       else Ok("flag not present")
+    case withMethod ->> Root / "resource" =>
```

Comment:
I kind of want to call it `resource`, but this probably overloads Cats-Effect too much.

Reply:
Instead of `withMethod` you mean?

Yeah, I also don't like it that `withMethod` has to be written twice, and that it feels like part of the DSL syntax but isn't.

Here's another idea - have the first item returned by the matcher be the entire request, then make `withMethod` (or maybe `respondByMethod`?) be an extension method on `Request`, so the usage looks something like
```scala
case req ->> Root / "foo" =>
  req.respondByMethod {
    case GET => ...
```
This also has the advantage of being more general, in that you can do arbitrary things with `req`. Obviously you can do that already by prefixing the matcher with `req @`, but that always feels kind of like a nasty workaround.